### ISSUE TITLE
[Bug]: Attempt to access an invalid index of column vector.

### ISSUE THREAD

**Issue by chrysanthemum-boy** [infiniflow/infinity#829]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Branch name

main

### Commit ID

c0fe904

### Actual behavior

When using the show_table function, an error occurs.
```
self = <infinity.remote_thrift.infinity_thrift_rpc.InfinityService.Client object at 0x74934f7bfa30>

    def recv_ShowDatabase(self):
        iprot = self._iprot
        (fname, mtype, rseqid) = iprot.readMessageBegin()
        if mtype == TMessageType.EXCEPTION:
            x = TApplicationException()
            x.read(iprot)
            iprot.readMessageEnd()
>           raise x
E           thrift.Thrift.TApplicationException: Attempt to access an invalid index of column vector: 3@src/storage/column_vector/column_vector.cpp:710
```

### Steps to reproduce

```Markdown
Function test case can be found in python/test/test_database.py `test_show_database`
```

**Comment by Ognimalf** [comment]
Fixed by #843 .

**Comment by chrysanthemum-boy** [comment]
It looks fixed.
